### PR TITLE
Generate and move to using DT_ prefixed FLASH defines

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: scripts/dts/extract/flash.py [hunk 1]
```diff
@@ -85,6 +85,9 @@ def _extract_partition(self, node_address):
 
         insert_defs("DT_FLASH_AREA", prop_def, prop_alias)
 
+    def _create_legacy_label(self, prop_alias, label):
+        prop_alias[label[3:]] = label
```
Comment (mbolivar) [inline scripts/dts/extract/flash.py]: Maybe `label.lstrip('DT_')` would be clearer?

Comment (galak) [reply scripts/dts/extract/flash.py]: Thanks, made the change.

**Review by mbolivar — APPROVED** [review]
Fine by me as long as the old defines are still around for v1.14

**Comment by galak** [comment]
> Fine by me as long as the old defines are still around for v1.14

They will be, I want to mark them with `__DEPRECATED_MACRO`

**Review by mbolivar — CHANGES_REQUESTED** [review]
@galak doing some testing and `west sign` was broken by 69c15f0ed1 (just 10 patches after it was merged...) and is still broken with these patches. I'm digging but want to -1 this for now until the problem is found

**Comment by galak** [comment]
> @galak doing some testing and `west sign` was broken by [69c15f0](https://github.com/zephyrproject-rtos/zephyr/commit/69c15f0ed17e1c74e1288b5bf7f99aeef829afeb) (just 10 patches after it was merged...) and is still broken with these patches. I'm digging but want to -1 this for now until the problem is found

Ok, at least 10 patches is a small amount to bisect :)

**Comment by mbolivar** [comment]
@galak this is not the root cause of west sign not working, but this PR has an issue. Take a look at the generated_dts_board.conf: 

https://hastebin.com/amedewumuv.ini

A lot of the 'legacy' defines are the string "None" instead of the number they should be.

**Comment by mbolivar** [comment]
> Ok, at least 10 patches is a small amount to bisect :)

That *is* the first bad patch; it was broken almost immediately haha

**Review by mbolivar — APPROVED** [review]
Retested the latest (2d2dfdb51401ddbc4d6348dcf3251129fe77a162) with https://github.com/JuulLabs-OSS/mcuboot/pull/405 applied. Looks good. Visually inspected the generated defines, including the legacy defines, and they are consistent with one another. Tested west sign still works.

**Comment by mbolivar** [comment]
recheck

**Pull request merged**